sysinit/coqargs.ml

```diff
@@ -226,11 +226,11 @@ let set_option = let open Goptions in function
   | opt, OptionSet (Some v) -> set_option_value ~locality:OptLocal (interp_set_option opt) opt v
   | opt, OptionAppend v -> set_string_option_append_value_gen ~locality:OptLocal opt v
 
-let get_compat_file = function
-  | "8.20" -> "Coq.Compat.Coq820"
-  | "8.19" -> "Coq.Compat.Coq819"
-  | "8.18" -> "Coq.Compat.Coq818"
-  | "8.17" -> "Coq.Compat.Coq817"
+let get_compat_files = function
+  | "8.20" -> ["Coq.Compat.Coq820"]
+  | "8.19" -> ["Coq.Compat.Coq819"]
+  | "8.18" -> ["Coq.Compat.Coq818"; "Ltac2.Compat818"]
+  | "8.17" -> ["Coq.Compat.Coq817"]
   | ("8.16" | "8.15" | "8.14" | "8.13" | "8.12" | "8.11" | "8.10" | "8.9" | "8.8" | "8.7" | "8.6" | "8.5" | "8.4" | "8.3" | "8.2" | "8.1" | "8.0") as s ->
```

Comment:
This code is autogenerated by
https://github.com/coq/coq/blob/c25a680f6ce9010a86ad9dbb7fe7f4b54e4521f1/dev/tools/update-compat.py#L242-L270
please update
https://github.com/coq/coq/blob/c25a680f6ce9010a86ad9dbb7fe7f4b54e4521f1/dev/tools/update-compat.py#L250
and
https://github.com/coq/coq/blob/c25a680f6ce9010a86ad9dbb7fe7f4b54e4521f1/dev/tools/update-compat.py#L258-L259
to deal with the new format.

Reply:
Probably a lot of this is because it's injecting updates to content in existing files that are only partially autogenerated.  It might make more sense to factor out the relevant files to be fully autogenerated, put the generating information in a different file (e.g., store the list of old versions in the generating script, not the generated file), maybe even move the relevant code into coq_rules.ml to be generated on the fly or w/e.

Reply:
Yes, I guess it would make sense to understand more about how compat is supposed to work in the roadmap before working too much on tooling; but indeed generating a self-contained file may be better.

For now I've side-stepped the issue by keeping the current Coq compat function, and doing the LTAC2 as a post-processing patch. This is IMO fine for now, WDYT?

Reply:
Seems fine, though it unfortunately adds a bit of technical debt rather than reducing the technical debt that I unfortunately introduced a while ago

Reply:
I wish I could have done better, but the change to the python code wasn't trivial at all.

IMHO we can implement a more flexible compat mechanism (and free of hacks) if we got the need. On the other hand, this case is pretty specific, as we need to decide:
- relationship between stdlib and ltac2
- if the status quo is kept, it is not clear how often Ltac2 compat code will appear on the wild, so it is possible that this hack is gone in 6 months.

Reply:
TBH I'm not sure the Ltac2 compat we have is any use, probably we could just delete it and forget about it.

Reply:
Indeed; IMHO the current compromise is OK tho.